### ISSUE TITLE
Booleans on both side

### ISSUE THREAD

**Issue by o5** [phpstan/phpstan-strict-rules#2]
What is wrong with shortcut `$doSomething && $something()`?

Full snippet:
```php
<?php

declare(strict_types = 1);

namespace Foo;

class Configurator
{
    public function loadConfig()
    {
        return $this;
    }
}

$configurator = new Configurator();

$devMode = true;
$devMode && $configurator->loadConfig();
```

Error:
```
------ -----------------------------------------------------------------------
  Line   bootstrap.php
 ------ -----------------------------------------------------------------------
  18     Only booleans are allowed in &&, Foo\Configurator given on
         the right side.
 ------ -----------------------------------------------------------------------
```

**Comment by ondrejmirtes** [comment]
To quote the readme:

> This repository contains additional rules that revolve around strictly and strongly typed code with no loose casting for those who want additional safety in extremely defensive programming.

I think that the error is self-explanatory. You should have `$devMode && $configurator->loadConfig() !== null;`. Or if you're using the expression as a side-effect and not as a condition in whose result you're interested in, then this would be better I think:

``php
if ($devMode) {
    $configurator->loadConfig();
}

**Comment by o5** [comment]
Yes, the error is self-explanatory and I fixed this issue yesterday with the solution as  you suggested as *better* :) But really don't see any risk with shortcut.

**Comment by ondrejmirtes** [comment]
The risk is there for anything else besides objects (things like empty string, empty array, `0`, string with `'0'` etc.) and I value consistency.
